src/main/java/seedu/address/model/commons/WanderlustDate.java

```diff
@@ -70,12 +71,51 @@ public boolean isAfter(WanderlustDate toCompare) {
         return value.isAfter(toCompare.getValue());
     }
 
+    /**
+     * Returns true if the start date is before or on the same day as end date.
+     */
+    public static boolean isValidStartAndEndDate(WanderlustDate startDate, WanderlustDate endDate) {
+        return startDate.getValue().compareTo(endDate.getValue()) <= 0;
+    }
 
+    /**
+     * Returns true if the date is set on today's date or after today's date.
+     */
+    public static boolean isDateAfterToday(WanderlustDate startDate) {
+        return startDate.getValue().compareTo(LocalDate.now()) >= 0;
+    }
 
+    /**
+     * Return number of days between {@code startDate} and {@code endDate} inclusive.
+     * @return Number of days between {@code startDate} and {@code endDate} inclusive.
+     */
+    public static long getNumOfDays(WanderlustDate startDate, WanderlustDate endDate) {
+        return ChronoUnit.DAYS.between(startDate.getValue(), endDate.getValue()) + 1;
+    }
+
+    /**
+     * Returns number of days and nights between the start and end date in a String format (e.g. "2D1N").
+     *
+     * @return Number of days (x) and nights (y) between the start and end date in the "xDyN" format.
+     */
+    public static String getNumOfDaysAndNights(WanderlustDate startDate, WanderlustDate endDate) {
```

Comment:
Ok this is good, clearer for the users also!

Reply:
okay! :)